### DISCUSSION TITLE
K8s环境下如何快速开发调测？

### DISCUSSION THREAD

**Discussion by messixukejia** [alibaba/ilogtail#288] [Help] [answered]
在开发K8s相关需求时
1、如何基于官方镜像包打开debug日志，并重启进程生效？
2、自己编的二进制文件、so，如何快速替换到容器里生效？

**Comment by yyuuttaaoo** [comment, marked as answer]
0、如果想要基于官方镜像包进行调试，那么需要用bash覆盖官方镜像的entrypoint，然后再跳入容器进行操作。
docker：`docker run -it --name docker_ilogtail -v /:/logtail_host:ro -v /var/run:/var/run  aliyun/ilogtail:edge bash` 
k8s：用command覆盖entrypoint
```yaml
       command:
            - sleep
            - 'infinity'
      # 删除livenessProbe，要不然会因为探测不到端口重启
```

1、如何基于官方镜像包打开debug日志，并重启进程生效？
开源ilogtail的pod在ilogtail namespace，通过`kubectl get pod -n ilogtail`可以找到对应pod
进入pod，`kubectl exec -it -n ilogtail xxx -- bash`
`ls`查看当前目录。
`apsara_log_conf.json`文件控制c++ core日志打印级别。替换`"AsyncFileSink" : "INFO"`为`"AsyncFileSink" : "DEBUG"`就能打开debug日志。
`plugin_logger.xml`文件控制go plugin日志打印。替换`minlevel="info"`为`minlevel="info"`就能打开debug日志。
修改日志级别后需重启ilogtail生效。
直接kill ilogtail后重新启动即可。`(./ilogtail) &`

2、自己编的二进制文件、so，如何快速替换到容器里生效？
由于ilogtail容器挂载了主机目录，因此将需要替换掉文件放到主机目录上容器内就能访问。
```bash
# 将开发机上编译的so scp到container所在node上
scp libPluginBase.so <user>@<node>:/home/<user>
```
主机的根路径在ilogtail容器中位于/logtail_host/，找到对应目录进行copy即可。
```bash
cp /logtail_host/home/<user>/libPluginBase.so /usr/local/ilogtail
```
